packages/datadog-instrumentations/src/http2/server.js

```diff
@@ -30,14 +30,12 @@ function wrapCreateServer (createServer) {
   }
 }
 
-function wrapResponseEmit (emit) {
-  const asyncResource = new AsyncResource('bound-anonymous-fn')
+function wrapResponseEmit (emit, ctx, store) {
   return function (eventName, event) {
-    return asyncResource.runInAsyncScope(() => {
-      if (eventName === 'close' && finishServerCh.hasSubscribers) {
-        finishServerCh.publish({ req: this.req })
-      }
-
+    storage('legacy').enterWith(store)
```

Comment:
The storage shouldn't be accessed from instrumentation. A new channel should be used instead which handles the storage on the subscriber side.

Reply:
Okay, I've removed the direct use of storage and added a new channel instead.